lib/block-supports/duotone.php

```diff
@@ -560,3 +572,44 @@ static function () use ( $filter_svg, $selector ) {
 // Remove WordPress core filter to avoid rendering duplicate support elements.
 remove_filter( 'render_block', 'wp_render_duotone_support', 10, 2 );
 add_filter( 'render_block', 'gutenberg_render_duotone_support', 10, 2 );
+
+add_action(
+	'wp_footer',
+	static function () {
+		// Get the presets from the theme.json.
+		$tree = WP_Theme_JSON_Resolver::get_merged_data();
+		$settings = $tree->get_settings();
+		$presets_by_origin = _wp_array_get( $settings, array( 'color', 'duotone' ), array() );
+		$flat_presets = [];
+		// Flatten the array
+		foreach( $presets_by_origin as $presets ) {
+			foreach( $presets as $preset ) {
+				$flat_presets[ _wp_to_kebab_case( $preset['slug'] ) ] = $preset;
+			}
+		}
+
+		foreach( WP_Duotone::$duotone_presets as $preset_slug ) {
+			// Convert $preset from a slug to an array of colors.
+			$duotone_preset = $flat_presets[ $preset_slug ];
+			$filter_svg = gutenberg_get_duotone_filter_svg( $duotone_preset );
+			echo $filter_svg;
+		}
+
+		/*
+		 * Safari renders elements incorrectly on first paint when the
+		 * SVG filter comes after the content that it is filtering, so
+		 * we force a repaint with a WebKit hack which solves the issue.
+		 */
+		global $is_safari;
+		if ( $is_safari ) {
+			/*
+			 * Simply accessing el.offsetHeight flushes layout and style
+			 * changes in WebKit without having to wait for setTimeout.
+			 */
+			printf(
+				'<script>( function() { var el = document.querySelector( %s ); var display = el.style.display; el.style.display = "none"; el.offsetHeight; el.style.display = display; } )();</script>',
+				wp_json_encode( $selector )
```

Comment:
`$selector` isn't defined in this function. Not sure what it's needed for, but it appears to already run through [the footer action above](https://github.com/WordPress/gutenberg/pull/48374/files#diff-fd626ca219f3d805667d480eabb2d890a5a8d2e015b37f03cf8980edf1ae3d58L522-L537) even if it's a global style, so maybe this Safari stuff isn't needed here?